src/appleseed/renderer/kernel/lighting/bdpt/bdptlightingengine.cpp

```diff
@@ -214,126 +249,60 @@ namespace
             return result;
         }
 
-        /// TODO:: precompute path density using fwd_pdf and rev_pdf.
-        /// in the final code, this function should be removed and the mis weight computation should be very simple.
-        /// example: https://github.com/mmp/pbrt-v3/blob/master/src/integrators/bdpt.cpp#L228
-        float compute_path_density(
+        // Compute the weight by considering hypothetical subpaths for s, t (Veach: 10.2).
+        // Modify the reverse pdf for the current vertex based on the connection strategy.
+        float compute_mis_weight(
             BDPTVertex*                 light_vertices,
             BDPTVertex*                 camera_vertices,
             const size_t                s,
-            const size_t                t,
-            const size_t                p,
-            const size_t                q)
+            const size_t                t)
         {
-            assert(p + q == s + t);
+            float sum_ri = 0.0f;  // The sum of the ratios of path densities for adjacent sampling strategies.
```

Comment:
End-of-line comments should not start with a capital nor end with a period (they're not really sentences, more annotations).